src/components/Modals/QrCode/Form.tsx

```diff
@@ -0,0 +1,153 @@
+
+  const handleQrSuccess = useCallback(
+    (decodedText: string) => {
+      ;(async () => {
+        // This should
+        // - Parse the address, amount and asset. This should also exhaust URI parsers (EVM and UTXO currently) and set the amount/asset if applicable
+        // - If there is a valid asset (i.e UTXO, or ETH, but not ERC-20s because they're unsafe), populates the asset and goes directly to the address step
+        // If no valid asset is found, it should go to the select asset step
+        const maybeUrlResult = await parseMaybeUrl({ value: decodedText })
+        methods.setValue(SendFormFields.AssetId, maybeUrlResult.assetId ?? '')
+        methods.setValue(SendFormFields.Input, decodedText.trim())
+        methods.setValue(SendFormFields.AssetId, maybeUrlResult.assetId ?? '')
+        if (maybeUrlResult.amountCryptoPrecision) {
+          const marketData = selectMarketDataById(store.getState(), maybeUrlResult.assetId ?? '')
+          methods.setValue(SendFormFields.CryptoAmount, maybeUrlResult.amountCryptoPrecision)
+          methods.setValue(
+            SendFormFields.FiatAmount,
+            bnOrZero(maybeUrlResult.amountCryptoPrecision).times(marketData.price).toString(),
+          )
+        }
+
+        // We don't parse EIP-681 URLs because they're unsafe
+        // Some wallets may be smart, like Trust just showing an address as a QR code to avoid dangerously unsafe parameters
+        // Others might do dangerous tricks in the way they represent an asset, using various parameters to do so
+        // There's also the fact that we will assume the AssetId to be the native one of the first chain we managed to validate the address
+        // Which may not be the chain the user wants to send, or they may want to send a token - so we should always ask the user to select the asset
+        if (maybeUrlResult.assetId === ethAssetId) return history.push(QrCodeRoutes.Select)
+        history.push(QrCodeRoutes.Address)
+      })()
+    },
+    [history, methods],
+  )
+
+  useEffect(() => {
+    history.push(QrCodeRoutes.Scan)
+  }, [history])
+
+  return (
+    <FormProvider {...methods}>
+      {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
+      <form onSubmit={methods.handleSubmit(handleFormSend)} onKeyDown={checkKeyDown}>
+        <AnimatePresence exitBeforeEnter initial={false}>
+          <Switch location={location} key={location.key}>
+            <Route path={QrCodeRoutes.Select}>
+              <SelectAssetRouter onBack={handleBack} onClick={handleAssetSelect} />
+            </Route>
+            <Route path={QrCodeRoutes.Address}>
+              <Address />
+            </Route>
+            <Route path={QrCodeRoutes.Details}>
+              <Details />
+            </Route>
+            <Route path={QrCodeRoutes.Scan}>
+              <QrCodeScanner onSuccess={handleQrSuccess} onBack={handleBack} />
+            </Route>
+            <Route path={SendRoutes.Confirm}>
```

Comment:
Not a typo - since we're leveraging the existing details component, confirm effectively pushes to the `/send/confirm` route

Reply:
With that being said, we can ditch the QR code routes entirely - 1728ec44522382227dff46680895f79141b5d838